src/main/java/de/tum/in/www1/artemis/service/connectors/AtheneService.java

```diff
@@ -0,0 +1,269 @@
+     * @param exerciseId the exerciseId to check for a running Athene task
+     * @return true, if a task for the given exerciseId is running
+     */
+    public boolean isTaskRunning(Long exerciseId) {
+        return runningAtheneTasks.contains(exerciseId);
+    }
+
+    /**
+     * Calls the remote Athene service to submit a Job for calculating automatic feedback
+     * @param exercise the exercise the automatic assessments should be calculated for
+     */
+    public void submitJob(TextExercise exercise) {
+        submitJob(exercise, 1);
+    }
+
+    /**
+     * Calls the remote Athene service to submit a Job for calculating automatic feedback
+     * Falls back to naive splitting for less than 10 submissions
+     * Note: See `TextSubmissionService:getTextSubmissionsByExerciseId` for selection of Submissions.
+     * @param exercise the exercise the automatic assessments should be calculated for
+     * @param maxRetries number of retries before the request will be canceled
+     */
+    public void submitJob(TextExercise exercise, int maxRetries) {
+        log.debug("Start Athene Service for Text Exercise \"" + exercise.getTitle() + "\" (#" + exercise.getId() + ").");
+
+        // Find all submissions for Exercise
+        List<TextSubmission> textSubmissions = textSubmissionService.getTextSubmissionsByExerciseId(exercise.getId(), true, false);
+
+        // We only support english languages so far, to prevent corruption of the clustering
+        textSubmissions.removeIf(textSubmission -> textSubmission.getLanguage() != Language.ENGLISH);
+
+        // Athene only works with 10 or more submissions
+        if (textSubmissions.size() >= 10) {
+
+            log.info("Calling Remote Service to calculate automatic feedback for " + textSubmissions.size() + " submissions.");
+
+            try {
+                final Request request = new Request(exercise.getId(), textSubmissions, ARTEMIS_SERVER_URL + ATHENE_RESULT_API_PATH + exercise.getId());
+                Response response = connector.invokeWithRetry(API_ENDPOINT, request, authorizationHeaderForSymmetricSecret(API_SECRET), maxRetries);
+                log.info("Remote Service to calculate automatic feedback responded: " + response.detail);
+
+                // Register task for exercise as running, AtheneResource calls finishTask on result receive
+                startTask(exercise.getId());
+            }
+            catch (NetworkingError networkingError) {
+                log.error("Error while calling Remote Service", networkingError);
+            }
+
+        }
+    }
+
+    /**
+     * Processes results coming back from the Athene system via callbackUrl (see AtheneResource)
+     * @param clusters the Map of calculated clusters to save to the database
+     * @param blocks the list of calculated textBlocks to save to the database
+     * @param exerciseId the exercise the automatic feedback suggestions were calculated for
+     */
+    @Transactional
```

Comment:
Can we remove transactional?

Reply:
Afaik not. We otherwise cause a `LazyInitializationException`

Reply:
Then you need to load all related objects from the database so that the operations can take place without such exceptions. We really would like to avoid @Transactional in such cases, because loading proxy objects from the database is quite performance intense